direct_chat.py

```
@@ -232,6 +232,124 @@ async def _handle_agent_mode(
+                if not pre.get("ok"):
+                    issues.append({
+                        "code": "git_repo_access",
+                        "message": f"Could not access repository at {repo_url}.",
+                        "fix_hint": "Verify the repository URL and ensure the GitHub token has access; ensure network egress to git hosts.",
+                        "details": {"error": pre.get("error")},
+                    })
+                # Branch/ref validation if provided in metadata
+                repo_ref = None
+                try:
+                    repo_ref = req.metadata.get("repo_ref") or req.metadata.get("branch") or req.metadata.get("ref") if req.metadata and isinstance(req.metadata, dict) else None
+                except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)